routers/web/repo/view.go

```diff
@@ -462,6 +462,16 @@ func renderFile(ctx *context.Context, entry *git.TreeEntry, treeLink, rawLink st
 			ctx.Data["HasSourceRenderedToggle"] = true
 		}
 
+		buf, _ := io.ReadAll(rd)
+
+		// empty: 0 lines; "a": one line; "a\n": two lines; "a\nb": two lines;
+		// the NumLines is only used for the display on the UI: "xxx lines"
+		if len(buf) == 0 {
+			ctx.Data["NumLines"] = 0
+		} else {
+			ctx.Data["NumLines"] = bytes.Count(buf, []byte{'\n'}) + 1
+		}
+
```

Comment:
Reading this here seems to break the markup rendering because the reader was already consumed, need a different solution to also count lines for markup files.

Reply:
Trying to count lines on the `output` returned from `markupRender` is also incorrect because we want source lines, not rendered lines. I wonder if it's possible to fix this without reading the file twice.

Reply:
One approach is to use a `LineCountReader` wrapper

Reply:
ok, will check how deep I have to alter to implement such a reader.